resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/kubernetes/KubernetesClusterSchedulerBackend.scala

```diff
@@ -368,14 +364,11 @@ private[spark] class KubernetesClusterSchedulerBackend(
    *         locality if an executor launches on the cluster node.
    */
   private def getNodesWithLocalTaskCounts() : Map[String, Int] = {
-    val executorPodsWithIPs = EXECUTOR_PODS_BY_IPS_LOCK.synchronized {
-      executorPodsByIPs.values.toList  // toList makes a defensive copy.
-    }
     val nodeToLocalTaskCount = mutable.Map[String, Int]() ++
       KubernetesClusterSchedulerBackend.this.synchronized {
         hostToLocalTaskCount
       }
-    for (pod <- executorPodsWithIPs) {
+    for ((_, pod) <- executorPodsByIPs) {
```

Comment:
Yes. Discovering executor pods is done only as a best effort. No need for consistency.